Source/Extensions/Blazorise.Charts/ScriptableValueBasedOptions.cs

```diff
@@ -0,0 +1,195 @@
+﻿#region Using directives
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+#endregion
+
+namespace Blazorise.Charts;
+
+/// <summary>
+/// Scriptable options also accept a function which is called for each of the underlying data values and that takes the unique
+/// argument context representing contextual information (see <see cref="ScriptableOptionsContext"/>).
+/// </summary>
+/// <typeparam name="TOptions">A value that is returned from a function.</typeparam>
+/// <typeparam name="TValue">The current label value of the dataset</typeparam>
+/// <typeparam name="TContext">A context representing contextual information.</typeparam>
+public class ScriptableValueBasedOptions<TOptions, TValue, TContext>
```

Comment:
Mostly copy pasted from [here](https://github.com/Megabit/Blazorise/blob/master/Source/Extensions/Blazorise.Charts/ScriptableOptions.cs) with additional support for TValue (the current dataset generated label)